### PR TITLE
Bug 1796133: Exclude fluent/elasticsearch pod from log collection and write logs to STDOUT

### PR DESCRIPTION
**Pull request by jcantrill**
Write fluent logs to stdout

ref: https://bugzilla.redhat.com/show_bug.cgi?id=1796133

### CODE REVIEW

**Comment by jcantrill** [comment]
cc @ewolinetz

File: pkg/generators/forwarding/fluentd/fluent_conf_test.go [hunk 1]
```diff
@@ -140,6 +140,7 @@ var _ = Describe("Generating fluentd config", func() {
 				@type tail
 				@id container-input
 				path "/var/log/containers/*.log"
+				exclude_path ['fluentd-*_openshift-logging_fluentd-*.log']
```
Comment (ewolinetz) [inline pkg/generators/forwarding/fluentd/fluent_conf_test.go]: nit, but is it necessary to have the container name in there too?
e.g. `'fluentd-*_openshift-logging_*.log'`

Comment (jcantrill) [reply pkg/generators/forwarding/fluentd/fluent_conf_test.go]: Corrected.

**Comment by jcantrill** [comment]
@ewolinetz updated to also exclude ES pods

File: pkg/generators/forwarding/fluentd/fluent_conf_test.go [hunk 2]
```diff
@@ -140,12 +140,12 @@ var _ = Describe("Generating fluentd config", func() {
 				@type tail
 				@id container-input
 				path "/var/log/containers/*.log"
+				exclude_path ['fluentd-*_openshift-logging_*.log', 'elasticsearch-*_openshift-logging_*.log']
```
Comment (ewolinetz) [inline pkg/generators/forwarding/fluentd/fluent_conf_test.go]: we should also exclude kibana, currently it writes to stdout and it can be noisy

Comment (jcantrill) [reply pkg/generators/forwarding/fluentd/fluent_conf_test.go]: done

File: pkg/generators/forwarding/fluentd/source.go [hunk 3]
```diff
@@ -24,8 +25,19 @@ func (engine *ConfigGenerator) generateSource(sources sets.String) (results []st
...
 		return results, fmt.Errorf("Unable to generate source configs for supported source types: %v", sources.List())
 	}
+	data := struct {
+		LoggingNamespace           string
+		CollectorPodNamePrefix     string
+		LogStorePodNamePrefix      string
+		VisualizationPodNamePrefix string
+	}{
+		constants.OpenshiftNS,
+		"fluentd",
```
Comment (ewolinetz) [inline pkg/generators/forwarding/fluentd/source.go]: nit, but should we make this a constant too for uniformity?

**Comment by ewolinetz** [comment]
tiny nit, otherwise :shipit:

**Comment by jcantrill** [comment]
@ewolinetz I also realized i can simply remove the file writing from the run script since it now lives here

File: files/fluentd/run.sh [hunk 4]
```diff
@@ -4,26 +4,8 @@ export MERGE_JSON_LOG=${MERGE_JSON_LOG:-false}
...
-    if [ ! -d $dirname ] ; then
-        mkdir -p $dirname
-    fi
-    touch $LOGGING_FILE_PATH; exec >> $LOGGING_FILE_PATH 2>&1
-
-    loggingargs="-o $LOGGING_FILE_PATH --log-rotate-age $LOGGING_FILE_AGE --log-rotate-size $LOGGING_FILE_SIZE"
-fi
 
-fluentdargs="--no-supervisor $loggingargs"
+fluentdargs="--no-supervisor"
```
Comment (ewolinetz) [inline files/fluentd/run.sh]: so i was actually just thinking about this... and this is probably a larger discussion but if we're logging to stdout how long does the container hold this information? and what happens to our running pod size the longer it lives? do we expect to just roll them periodically? or can we configure it to age out older stdout entries?

Comment (jcantrill) [reply files/fluentd/run.sh]: > so i was actually just thinking about this... and this is probably a larger discussion and not a blocker but if we're logging to stdout how long does the container hold this information? and what happens to our running pod size the longer it lives? do we expect to just roll them periodically? or can we configure it to age out older stdout entries?

I believe this is the responsibility of the runtime to manage the logfile and its rotation

**Pull request merged**